bot/exts/recruitment/talentpool/_cog.py

```diff
@@ -342,18 +342,75 @@ async def end_nomination_command(self, ctx: Context, user: MemberOrUser, *, reas
+    @has_any_role(*STAFF_ROLES)
     async def nomination_edit_group(self, ctx: Context) -> None:
         """Commands to edit nominations."""
         await ctx.send_help(ctx.command)
 
     @nomination_edit_group.command(name='reason')
-    @has_any_role(*MODERATION_ROLES)
-    async def edit_reason_command(self, ctx: Context, nomination_id: int, actor: MemberOrUser, *, reason: str) -> None:
-        """Edits the reason of a specific nominator in a specific active nomination."""
+    @has_any_role(*STAFF_ROLES)
+    async def edit_reason_command(
+        self,
+        ctx: Context,
+        nominee_or_nomination_id: Union[UnambiguousMemberOrUser, int],
+        nominator: Optional[UnambiguousMemberOrUser] = None,
+        *,
+        reason: str
+    ) -> None:
+        """
+        Edit the nomination reason of a specific nominator for a given nomination.
+
+        If nominee_or_nomination_id resolves to a member or user, edit the currently active nomination for that person.
+        Otherwise, if it's an int, look up that nomination ID to edit.
+
+        If no nominator is specified, assume the invoker is editing their own nomination reason.
+        Otherwise, edit the reason from that specific nominator.
+
+        Raise a permission error if a non-mod staff member invokes this command on a
+        specific nomination ID, or with an nominator other than themselves.
+        """
+        # If not specified, assume the invoker is editing their own nomination reason.
+        nominator = nominator or ctx.author
+
+        if not any(role.id in MODERATION_ROLES for role in ctx.author.roles):
+            if ctx.channel.id != Channels.nominations:
+                await ctx.send(f":x: Nomination edits must be run in the <#{Channels.nominations}> channel")
+                return
+
+            if nominator != ctx.author or isinstance(nominee_or_nomination_id, int):
+                # Invoker has specified another nominator, or a specific nomination id
+                raise BadArgument(
+                    "Only moderators can edit specific nomination IDs, "
+                    "or the reason of a nominator other than themselves."
+                )
+
+        await self._edit_nomination_reason(
+            ctx,
+            target=nominee_or_nomination_id,
+            actor=nominator,
+            reason=reason
+        )
+
+    async def _edit_nomination_reason(
+        self,
+        ctx: Context,
+        *,
+        target: Union[int, Member],
+        actor: MemberOrUser,
+        reason: str,
+    ) -> None:
+        """Edit a nomination reason in the database after validating the input."""
         if len(reason) > REASON_MAX_CHARS:
-            await ctx.send(f":x: Maxiumum allowed characters for the reason is {REASON_MAX_CHARS}.")
+            await ctx.send(f":x: Maximum allowed characters for the reason is {REASON_MAX_CHARS}.")
             return
+        if isinstance(target, Member):
```

Comment:
The command can resolve to a user. What happens if you try to nominate someone off-server?

Reply:
It'll just work, I'll swap this around, so it check isinstance of int and then an else to catch Member & User

Reply:
https://github.com/python-discord/bot/pull/1841/commits/d444ebd8c3f9b10e021ecc94da8bbac482a1b236